packages/block-editor/src/components/editor-styles/index.js

```diff
@@ -67,17 +67,35 @@ function useDarkThemeBodyClassName( styles, scope ) {
 	);
 }
 
-function EditorStyles( { styles, scope } ) {
-	const overrides = useSelect(
+function getStyleOverrideById( overrides, id ) {
+	if ( ! overrides?.length || ! id ) {
+		return [];
+	}
+
+	return overrides.find( ( [ _id ] ) => _id === id ) || [];
+}
+
+function EditorStyles( { styles, scope, overrides } ) {
+	const styleOverrides = useSelect(
```

Comment:
If I'm understanding this correctly, this was already selected if you implement `overrides`. So we don't need to select it here again if `overrides` is defined, and we don't need to merge both?

Reply:
What I'm seeing is that `useUpdateBlockStyleVariationOverridesWithConfig` uses `unlock( select( blockEditorStore ) ).getStyleOverrides()` from the store, then merges it with something else, then again we get `unlock( select( blockEditorStore ) ).getStyleOverrides()` and they are merged. So there's one merge too much here?